src/Libraries/DesignScriptBuiltin/Dictionary.cs

```diff
@@ -59,9 +59,11 @@ private Dictionary(ImmutableDictionary<string, object> dict)
             ///     Produces the values in a Dictionary.
             /// </summary>
             /// <returns name="values">Values of the dictionary</returns>
-            // [AllowRankReduction]
-            // TODO: Consider adding this attribute in 3.0 (DYN-1697)
-            public IEnumerable<object> Values => D.Values;
+            public IEnumerable<object> Values
+            {
+                [return: ArbitraryDimensionArrayImport]
```

Comment:
`ArbitraryDimensionArrayImport` and `AllowRankReduction` attributes are slightly different in behaviour. The first one can be used for function parameters and return values and allows arbitrary ranked arrays and mixed rank arrays from being passed and returned to and from functions. The latter attribute can only be applied to _return_ value of a function. It returns a single value in the case that a function returning a list returns a list of a single value (reduces the rank of a return value).

The thing that both attributes are similar in is that both of them convert the value to arbitrary rank, if it is an array (and if it's an array containing more than one value in the case of `AllowRankReduction`).

In this case, we don't want to change the behaviour of `Dictionary.Values` as far as possible, so we don't want to use `AllowRankReduction` but use `ArbitraryArrayDimensionImport` attribute instead so we can support returning arbitrarily ranked arrays.

Reply:
`ArbitraryArrayDimensionImport` in the past even though it could be applied to function return values and function parameters, we didn't add support to make it handle return values. So, in this PR, I've added that support as well as applied it to `Dictionary.Values`.

Reply:
Sounds good. And this approach will not break the API?

Reply:
I think this is fine, can you name the syntax though, my googling is coming up short - `[return: ArbitraryDimensionArrayImport]`